lib/rspec/support/caller_filter.rb

```diff
@@ -73,10 +73,24 @@ def self.first_non_rspec_line(skip_frames=3, increment=5)
         end
       end
     else
+      CallerFinder = Class.new(StandardError)
+      class LazyBacktrace
+        def initialize(exception)
+          @exception = exception
+        end
+
+        def to_s
+          @exception.backtrace.find { |line| line !~ IGNORE_REGEX }
+        end
+      end
       # Earlier rubies do not support the two argument form of `caller`. This
       # fallback is logically the same, but slower.
       def self.first_non_rspec_line(*)
-        caller.find { |line| line !~ IGNORE_REGEX }
+        begin
+          raise CallerFinder
+        rescue CallerFinder => error
+          return LazyBacktrace.new(error)
+        end
```

Comment:
> I've just benchmarked raise against caller and caller_location and added that to this PR and it seems that although a bare raise and capture is faster than both a bare caller or a bare caller_location, it isn't faster than the skip frames example for caller_locations?

Looks like you're right.  This surprises me!  At first I thought your benchmark was flawed in that the backtrace sizes were so small and with a larger, more realistic trace (e.g. 50 or 100 frames) it would be faster to use `raise`.  Turns out I was wrong -- so the commit I just pushed.

Given that all non-EOL'd versions of ruby have `caller_locations`, I think we shouldn't make a change here.  It's not worth it to me to add complexity to optimize EOL'd versions of ruby, which is all we can do here.  I do think it's worth pushing the benchmark to `master` so we have the record of it in the future.